content/exchange/artifacts/Generic.VSCode.Extensions.yaml

```diff
@@ -0,0 +1,84 @@
+name: Generic.VSCode.Extensions
+author: "Andreas Misje – @misje"
+description: |
+  List all installed Visual Studio code extensions
+
+  This artifact lists all installed VS Code exensions on all supported
+  operating systems. This list can be used to hunt for known malicious
+  extensions by name or ID.
+
+  The artifact returns the raw JSON from the extensions file. The notebook
+  provides a more user-friendly view.
+
+type: CLIENT
+
+parameters:
+  - name: ExtensionGlobs
+    description: |
+      Where to look for extension.json files on varios OSes
+    type: csv
+    default: |
+      OS,Glob
+      windows,C:\Users\*\.vscode\extensions\extensions.json
+      linux,/home/*/.vscode/extensions/extensions.json
+      darwin,/Users/*/.vscode/extensions/extensions.json
+
+column_types:
+  - name: ID
+    description: Extension name
+  - name: _UUID
+    description: Extension unique identifier
+  - name: Version
+    description: Extension version
+  - name: UpToDate
+    description: Whether VSCode considers this version up-to-date
+  - name: Publisher
+    description: The name of the publisher
+  - name: _PublisherID
+    description: Publisher unique identifier
+  - name: InstalledAt
+    type: timestamp
+    description: When the extension was installed
+  - name: _Location
+    description: Local path to extension files
+  - name: _Info
+    description: All available information about the extension
+
+sources:
+  - query: |
+      LET Info <= SELECT OS
+        FROM info()
+      LET Globs = SELECT Glob
+        FROM ExtensionGlobs
+        WHERE OS = Info[0].OS
+      SELECT *
+      FROM foreach(
+        row={
+          SELECT OSPath
+          FROM glob(globs=Globs.Glob)
+        },
+        query={
+          SELECT OSPath,
+                 parse_json_array(data=Data) AS Extensions
+          FROM read_file(filenames=OSPath)
```

Comment:
Would it be better to just upload the files rather than include them in the JSON blob? We try to avoid that as it can explode the size of the JSON responses and lead to issues in the GUI.

You can use preview_upload column type to preview the file in the GUI as well which makes it a lot nicer